### PR TITLE
chore(helper-plugin)!: remove form

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* removes the Form component from the helper-plugin

### Why is it needed?

* we can either use Formik directly, or preferably use the Form from the admin

### CODE REVIEW

File: packages/core/helper-plugin/MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -17,6 +17,20 @@ import { DateTimePicker } from '@strapi/helper-plugin';
...
+
+```ts
+// Before
+import { Form } from '@strapi/helper-plugin';
+
+// After
+import { Form } from '@strapi/strapi/admin';
+```
+
+Users should note that any use of the Formik library will no longer work & insted should look at the documentation for the `Form` component.
```
Comment (Marc-Roig) [inline packages/core/helper-plugin/MIGRATION_GUIDE.md:32]: should this be mentioned in the breaking changes list https://www.notion.so/strapi/831ef9cf0d66425a9f97d4981f96c1cc?v=2ee276ead71b41adb17ef477a47c96ac ?

**Review by Marc-Roig — APPROVED** [review]

File: packages/core/helper-plugin/MIGRATION_GUIDE.md (hunk 1 above)
Comment (joshuaellis) [reply packages/core/helper-plugin/MIGRATION_GUIDE.md:32]: I'll probably write a section on the helper-plugin being removed and link to this migration guide but yes

File: packages/core/admin/admin/src/pages/Settings/pages/Users/EditPage.tsx [hunk 2]
```diff
@@ -177,13 +175,10 @@ const EditPage = () => {
...
   } satisfies InitialData;
 
-  /**
-   * TODO: Convert this to react-query.
-   */
   const handleSubmit = async (body: InitialData, actions: FormikHelpers<InitialData>) => {
     lockApp?.();
 
-    const { confirmPassword, password, ...bodyRest } = body;
+    const { confirmPassword: _confirmPassword, password, ...bodyRest } = body;
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/pages/Settings/pages/Users/EditPage.tsx:181]: Do we need _confirmPassword ?

File: packages/core/admin/admin/src/pages/Auth/components/ForgotPassword.tsx [hunk 3]
```diff
@@ -62,42 +63,41 @@ const ForgotPassword = () => {
               }
             }}
             validationSchema={yup.object().shape({
-              email: yup.string().email(translatedErrors.email).required(translatedErrors.required),
+              email: yup
+                .string()
+                .email({
+                  id: translatedErrors.email,
+                  defaultMessage: 'This is not a valid email.',
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/pages/Auth/components/ForgotPassword.tsx:70]: Should we update this here? https://github.com/strapi/strapi/blob/56142c70f8b51d644d00a5ed70f1795637b29d91/packages/core/email/admin/src/translations/en.json#L22

File: packages/core/admin/admin/src/pages/Settings/pages/Users/EditPage.tsx (hunk 2 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/pages/Settings/pages/Users/EditPage.tsx:181]: No we want to exclude it from the API, eslint throws a warning for unused variables if they're not prefixed with _

File: packages/core/admin/admin/src/pages/Auth/components/Login.tsx [hunk 4]
```diff
@@ -103,90 +111,43 @@ const Login = ({ children }: LoginProps) => {
...
+                  }),
+                  required: true,
+                  type: 'string' as const,
+                },
+                {
+                  label: formatMessage({
+                    id: 'global.password',
+                    defaultMessage: 'Password',
+                  }),
+                  name: 'password',
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/pages/Auth/components/Login.tsx:132]: I think we need to mark this as required? That is one of the failing FE tests packages/core/admin/admin/src/pages/Auth/components/tests/Login.test.tsx

Comment (joshuaellis) [reply packages/core/admin/admin/src/pages/Auth/components/Login.tsx:132]: ⭐

**Pull request merged**